posthog/temporal/tests/batch_exports/test_http_batch_export_workflow.py

```diff
@@ -0,0 +1,466 @@
+from aiohttp.client_exceptions import ClientResponseError
+import asyncio
+import datetime as dt
+import json
+from random import randint
+from uuid import uuid4
+
+from aioresponses import aioresponses
+import pytest
+import pytest_asyncio
+from django.conf import settings
+from django.test import override_settings
+from temporalio import activity
+from temporalio.client import WorkflowFailureError
+from temporalio.common import RetryPolicy
+from temporalio.testing import WorkflowEnvironment
+from temporalio.worker import UnsandboxedWorkflowRunner, Worker
+
+from posthog.temporal.batch_exports.batch_exports import (
+    create_export_run,
+    iter_records,
+    update_export_run_status,
+)
+from posthog.temporal.batch_exports.clickhouse import ClickHouseClient
+from posthog.temporal.batch_exports.http_batch_export import (
+    HttpBatchExportInputs,
+    HttpBatchExportWorkflow,
+    HttpInsertInputs,
+    insert_into_http_activity,
+    http_default_fields,
+)
+from posthog.temporal.tests.utils.events import generate_test_events_in_clickhouse
+from posthog.temporal.tests.utils.models import (
+    acreate_batch_export,
+    adelete_batch_export,
+    afetch_batch_export_runs,
+)
+
+pytestmark = [
+    pytest.mark.asyncio,
+    pytest.mark.django_db,
+]
+
+TEST_URL = "http://example.com/batch"
+TEST_TOKEN = "abcdef123456"
+
+
+@pytest.fixture
+def http_config():
+    return {
+        "url": TEST_URL,
+        "token": TEST_TOKEN,
+    }
+
+
+class MockServer:
+    def __init__(self):
+        self.records = []
+
+    def post(self, url, data, **kwargs):
+        data = json.loads(data.read())
+        assert data["api_key"] == TEST_TOKEN
+        self.records.extend(data["batch"])
```

Comment:
For a more realistic scenario, couldn't we fire these requests to the real capture running in local stack?

Reply:
It's certainly possible. I like the speed of mocked tests (for development anyway). The batch API is pretty simple and the schema is hardcoded for migrations, I'd rather leave this for if we make this more complex.